In [1]:
import pandas as pd

In [3]:
df = pd.read_csv('../data/chess_games.csv')

In [4]:
df.describe()

game_id         turns  white_rating  black_rating  opening_moves
count  20058.000000  20058.000000  20058.000000  20058.000000   20058.000000
mean   10029.500000     60.465999   1596.631868   1588.831987       4.816981
std     5790.390185     33.570585    291.253376    291.036126       2.797152
min        1.000000      1.000000    784.000000    789.000000       1.000000
25%     5015.250000     37.000000   1398.000000   1391.000000       3.000000
50%    10029.500000     55.000000   1567.000000   1562.000000       4.000000
75%    15043.750000     79.000000   1793.000000   1784.000000       6.000000
max    20058.000000    349.000000   2700.000000   2723.000000      28.000000

In [5]:
df.head(5)

game_id  rated  turns victory_status winner time_increment       white_id  \
0        1  False     13    Out of Time  White           15+2       bourgris   
1        2   True     16         Resign  Black           5+10           a-00   
2        3   True     61           Mate  White           5+10         ischia   
3        4   True     61           Mate  White           20+0  daniamurashov   
4        5   True     95           Mate  White           30+3      nik221107   

   white_rating      black_id  black_rating  \
0          1500          a-00          1191   
1          1322     skinnerua          1261   
2          1496          a-00          1500   
3          1439  adivanov2009          1454   
4          1523  adivanov2009          1469   

                                               moves opening_code  \
0  d4 d5 c4 c6 cxd5 e6 dxe6 fxe6 Nf3 Bb4+ Nc3 Ba5...          D10   
1  d4 Nc6 e4 e5 f4 f6 dxe5 fxe5 fxe5 Nxe5 Qd4 Nc6...          B00   
2  e4 e5 d3 d6 Be3 c6 Be2 b5 Nd2 a5 a4 c5 axb5 Nc...          C20   
3  d4 d5 Nf3 Bf5 Nc3 Nf6 Bf4 Ng4 e3 Nc6 Be2 Qd7 O...          D02   
4  e4 e5 Nf3 d6 d4 Nc6 d5 Nb4 a3 Na6 Nc3 Be7 b4 N...          C41   

   opening_moves                        opening_fullname    opening_shortname  \
0              5        Slav Defense: Exchange Variation         Slav Defense   
1              4  Nimzowitsch Defense: Kennedy Variation  Nimzowitsch Defense   
2              3   King's Pawn Game: Leonardis Variation     King's Pawn Game   
3              3  Queen's Pawn Game: Zukertort Variation    Queen's Pawn Game   
4              5                        Philidor Defense     Philidor Defense   

  opening_response    opening_variation  
0              NaN   Exchange Variation  
1              NaN    Kennedy Variation  
2              NaN  Leonardis Variation  
3              NaN  Zukertort Variation  
4              NaN                  NaN

In [6]:
df.columns

Index(['game_id', 'rated', 'turns', 'victory_status', 'winner',
       'time_increment', 'white_id', 'white_rating', 'black_id',
       'black_rating', 'moves', 'opening_code', 'opening_moves',
       'opening_fullname', 'opening_shortname', 'opening_response',
       'opening_variation'],
      dtype='object')

# Inspección de datos

- game_id (int): Identificador único de la partida.
- rated (bool): Si la partida fue clasificada o no (True, False).
- turns (int): Número de turnos en la partida. 
- victory_status (string): Estado de la partida (Out of Time, Resign, Mate, Draw).
- winner (string): Ganador de la partida (White, Black, Draw).
- time_increment (string): Duración inicial de la partida + incremento de tiempo en segundos, Ej: 15+2 => Partida con duración base de 15 minutos + 2 segundos de adición por movimiento.
- white_id (string): ID del jugador blanco.
- white_rating (int): Rating del jugador blanco.
- black_id (string): ID del jugador negro.
- black_rating (int): Rating del jugador negro.
- moves (string): Todos los movimientos realizados en la partida.
- opening_code (string): Código ECO usado para categorizar la apertura.
- opening_moves (int): Número de movimientos realizados en la apertura. 
- opening_fullname (string): Nombre completo de la apertura realizada.
- opening_shortname (string): Nombre abreviado de la apertura.
- opening_response (string): Respuesta por parte de las negras a la apertura (Refused, Accepted, Declined)
- opening_variation (string): Variación de la apertura.

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20058 entries, 0 to 20057
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   game_id            20058 non-null  int64 
 1   rated              20058 non-null  bool  
 2   turns              20058 non-null  int64 
 3   victory_status     20058 non-null  object
 4   winner             20058 non-null  object
 5   time_increment     20058 non-null  object
 6   white_id           20058 non-null  object
 7   white_rating       20058 non-null  int64 
 8   black_id           20058 non-null  object
 9   black_rating       20058 non-null  int64 
 10  moves              20058 non-null  object
 11  opening_code       20058 non-null  object
 12  opening_moves      20058 non-null  int64 
 13  opening_fullname   20058 non-null  object
 14  opening_shortname  20058 non-null  object
 15  opening_response   1207 non-null   object
 16  opening_variation  14398 non-null  objec

In [8]:
df.isna().sum()

game_id                  0
rated                    0
turns                    0
victory_status           0
winner                   0
time_increment           0
white_id                 0
white_rating             0
black_id                 0
black_rating             0
moves                    0
opening_code             0
opening_moves            0
opening_fullname         0
opening_shortname        0
opening_response     18851
opening_variation     5660
dtype: int64

In [28]:
df['opening_variation'].unique()

array(['Exchange Variation', 'Kennedy Variation', 'Leonardis Variation',
       'Zukertort Variation', nan, 'Mongoose Variation',
       'Pietrowsky Defense', 'Schilling-Kostic Gambit',
       'Mieses-Kotroc Variation', 'Advance Variation', 'Knight Variation',
       '#2', 'Italian Variation', 'Two Knights Defense', '#3',
       'Anti-Fried Liver Defense', 'Bowdler Attack',
       "King's English Variation", 'Smith-Morra Gambit #2',
       'Chigorin Variation', 'Haxo Gambit', 'Marshall Defense',
       'Canal Attack', 'Central Variation', 'Tartakower Variation',
       'Dragon Variation', 'Closed Variation', 'Normal Variation',
       'Beyer Gambit', 'Ross Gambit', 'Mengarini Variation',
       'Larsen Variation', 'Lopez Countergambit', 'Kingside Fianchetto',
       "Queen's Knight", 'Mason Attack', 'Advance', 'London System',
       'La Bourdonnais Variation', 'Anglo-Lithuanian Variation',
       'Winawer Variation', 'Steinitz Countergambit', 'Giuoco Pianissimo',
       'Marshall Vari

In [9]:
df.duplicated().sum()

0

- Solo existen 1207 datos en opening_response, el resto son 18851 datos nulos.
- opening_variation solo posee 14398 datos completos, el resto son 5660 datos nulos, ademas, existe una opening variation que es #3.